public/js/replay.js

```diff
@@ -70,23 +94,169 @@ function Replay (params) {
    *  Add any flagged versions to the replay slider div
    */
   this.addFlags = function() {
+    this.flagged = [];
+    this.current_flag = null;
+    $('#flag-left').prop("disabled",true);
     for (var i = 0; i < instance.operations.length; i++) {
       if (instance.operations[i].flagged) {
+        this.flagged.push(i+1);
         var percentLeft = (instance.operations[i].v / instance.operations.length * 100);
         var element = flagTemplate.replace('{{percentLeft}}', percentLeft);
         $('#replaySlider').append (element);
       }
     }
+    if (this.flagged.length === 0){
+      $('#flag-left').prop("disabled",true);
+      $('#flag-right').prop("disabled",true);
+    }
+  };
+
+  /*
+  * Non-anonymous version of the slider listener. Just a copy-paste of it really..
+  */
+  this.setVersion = function (version) {
```

Comment:
Rather than having the same code twice in the file, pass this function as the callback to the slider's formatter handler.

That will make it much easier to maintain.